apps/meteor/app/importer-slack/server/importer.js

```diff
@@ -235,7 +237,7 @@ export class SlackImporter extends Base {
 			// If we have no slack message yet, then we can insert them instead of upserting
 			this._useUpsert = !Messages.findOne({ _id: /slack\-.*/ });
 
-			zip.forEach((entry) => {
+			const operations = zip.map(async (entry) => {
```

Comment:
I'll run some stress tests on this to ensure this change doesn't affect the operation. The zip loop is a very sensitive part of the import process.

Reply:
Go there. I was thinking on changing this to either this async map or a for await of, let me know your results 🤗